JOSESwift/Sources/CryptoImplementation/PBES2.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  PBES2.swift
+//  JOSESwift
+//
+//  Created by Tobias Hagemann on 07.12.23.
+//
+//  ---------------------------------------------------------------------------
+//  Copyright 2023 Airside Mobile Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//  ---------------------------------------------------------------------------
+//
+
+import Foundation
+import CommonCrypto
+
+internal enum PBES2Error: Error {
+    case unknownOrUnsupportedAlgorithm
+}
+
+internal struct PBES2 {
+    static let defaultIterationCount = 1_000_000
```

Comment:
A million iterations feels arbitrary. Is there any reason for picking exactly this number? If not we could go with the [recommended minimum](https://datatracker.ietf.org/doc/html/rfc7518#section-4.8.1.2) of `1_000`. thoughts?

Reply:
It has already been merged, but I wanted to mention that it's probably better to set a "secure" default than to set a minimum, defined in 2015.

1 million is probably "arbitrary", but OWASP recommends iteration counts between 210k and 1.3 million, see: https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#pbkdf2

1000 is far too low for today's standards. I would argue that this wouldn't be a good "default" value.